PlayerPanel.js

```diff
@@ -34,87 +34,121 @@ function update_pclist() {
 
 
 	});
-
-
-
+	const addPartyButtonContainer = $("<div class='add-party-container'></div>");
+	const addPartyButton = $("<button id='add-party'>ADD PARTY</button>");
+	addPartyButton.on('click', () => {
+		window.pcs.forEach(function (p, i) {
+			token_button({ target: $(`[data-name='${p.name}']`) }, i, window.pcs.length);

```

Comment:
Couple gotcha issues here.

What happens if two  items of the same name are in the List? E.G. Players Bob and Bob? Looks like the selector pulls from the name, so we would have a duplicate here. I don't know what the behavior is. To solve this, we should add a key here (assuming this doesn't impact the UI) so the selection is unique. We can do that by appending the index to the name, or preferably using another UUID (if something like that exists on this model).

Additionally, what happens when an upstream bug causes name to be undefined?

There is a trick here to avoid a nasty `undefined` or error if `p` or `p.name` is undefined.

```
window.pcs.forEach(function ({ name = '' }, i) {
    token_button({ target: $(`[data-name='${name}']`) }, i, window.pcs.length);
    // . . .
 ```

Reply:
good catch, didn't thought of that use case, fixed.